unitylibs/core/workflow/workflow-acrobat/upload-handler.js

```diff
@@ -46,82 +46,51 @@ export default class UploadHandler {
     return blob;
   }
 
-  async uploadFileToUnityWithRetry(url, blobData, fileType, assetId, signal, chunkNumber = 0) {
-    let retryDelay = 1000;
-    const maxRetries = 4;
-    let error = null;
-    for (let attempt = 1; attempt <= maxRetries; attempt++) {
-      try {
-        const response = await this.uploadFileToUnity(url, blobData, fileType, assetId, signal, chunkNumber);
-        if (response.ok) {
-          this.actionBinder.dispatchAnalyticsEvent('chunk_uploaded', {
-            chunkUploadAttempt: attempt,
-            assetId,
-            chunkNumber,
-            size: `${blobData.size}`,
-            type: `${fileType}`,
-          });
-          return { response, attempt };
-        }
-      } catch (err) {
-        if (err.name === 'AbortError') throw err;
-        error = err;
-      }
-      if (attempt < maxRetries) {
-        const delay = retryDelay;
-        await new Promise((resolve) => { setTimeout(resolve, delay); });
-        retryDelay *= 2;
-      }
+  async afterUploadFileToUnity(assetId, blobData, chunkNumber, fileType, response, attempt) {
+    if (response.ok) {
+      this.actionBinder.dispatchAnalyticsEvent('chunk_uploaded', {
+        chunkUploadAttempt: attempt,
+        assetId,
+        chunkNumber,
+        size: `${blobData.size}`,
+        type: `${fileType}`,
+      });
+      return response;
     }
-    if (error) error.message += ', Max retry delay exceeded during upload';
-    else error = new Error('Max retry delay exceeded during upload');
+    const error = new Error(response.statusText || 'Upload request failed');
+    error.status = response.status;
+    await this.actionBinder.dispatchErrorToast('upload_warn_chunk_upload', response.status, `Failed when uploading chunk to storage; ${response.statusText}, ${assetId}, ${blobData.size} bytes`, true, true, {
+      code: 'upload_warn_chunk_upload',
+      subCode: chunkNumber,
+      desc: `Failed when uploading chunk to storage; ${response.statusText}, ${assetId}, ${blobData.size} bytes; status: ${response.status}`,
+    });
+    if (attempt < this.actionBinder.workflowCfg.targetCfg.fetchApiConfig.default.retryParams.maxRetries) return response;
```

Comment:
Would it make sense to cache this config value if it's being looked up multiple times?